consul/check.py

```diff
@@ -246,18 +246,27 @@ def check(self, instance):
             # Make service checks from health checks for all services in catalog
             health_state = self.consul_request(instance, '/v1/health/state/any')
 
+            sc = {}
+            # compute the highest status level (OK < WARNING < CRITICAL) a a check among all the nodes is running on.
             for check in health_state:
+                sc_id = '{0}-{1}-{2}'.format(check['CheckID'], check.get('ServiceID', ''), check.get('ServiceName', ''))
                 status = self.STATUS_SC.get(check['Status'])
                 if status is None:
                     continue
 
-                tags = ["check:{0}".format(check["CheckID"])]
-                if check["ServiceName"]:
-                    tags.append("service:{0}".format(check["ServiceName"]))
-                if check["ServiceID"]:
-                    tags.append("consul_service_id:{0}".format(check["ServiceID"]))
+                if sc_id not in sc:
+                    tags = ["check:{0}".format(check["CheckID"])]
+                    if check["ServiceName"]:
+                        tags.append("service:{0}".format(check["ServiceName"]))
+                    if check["ServiceID"]:
+                        tags.append("consul_service_id:{0}".format(check["ServiceID"]))
+                    sc[sc_id] = {'status': status, 'tags': tags}
 
-                self.service_check(self.HEALTH_CHECK, status, tags=main_tags+tags)
+                elif status > sc[sc_id]['status']:
```

Comment:
I agree. And since `AgentCheck.UNKNOWN` is greater than `AgentCheck.CRITICAL`  but we want to handle it as the least priority it makes even more sense.